src/tensor_ops/broadcast.rs

```diff
@@ -0,0 +1,265 @@
+//! TODO describe implementation details of why there are 3 versions of the traits and how they differ.
```

Comment:
It's pretty confusing to see multiple versions of broadcast functions in ide completion. Ideally people should NOT be using ConstBroadcast or BroadcastTo, however those exist to make generic functions possible.

BroadcastTo exists for some of *_axis methods that need to immediately broadcast reduced values back to the original size (std_axis, normalize, log_softmax).

ConstBroadcast exists to specify the associated type that is used to implement Broadcast[1-4].

Reply:
BroadcastTo could potentially be removed if variants of the _axis functions exist that don't actually reduce or change dimension. e.g. `sum_axis_rebroadcast<T, const I: isize>(t: T) -> T`

Reply:
Or another option is to remove Broadcast[1-4] in favor of using BroadcastTo everywhere. I think this option seems feasible

Reply:
Removed everything in favor of juts BroadcastTo, which is the "opposite" of the current Reduce1